test/test_conn_monitor.cpp

```diff
@@ -0,0 +1,121 @@
+/* Copyright (c) 2018-2022 Marcelo Zimbres Silva (mzimbres@gmail.com)
+ *
+ * Distributed under the Boost Software License, Version 1.0. (See
+ * accompanying file LICENSE.txt)
+ */
+
+#include <boost/redis/connection.hpp>
+#include <boost/redis/ignore.hpp>
+#include <boost/redis/request.hpp>
+#include <boost/redis/response.hpp>
+
+#include <boost/asio/error.hpp>
+#include <boost/core/lightweight_test.hpp>
+
+#include "common.hpp"
+
+#include <cstddef>
+
+namespace net = boost::asio;
+using boost::system::error_code;
+using boost::redis::connection;
+using boost::redis::request;
+using boost::redis::ignore;
+using boost::redis::operation;
+using boost::redis::generic_response;
+using boost::redis::consume_one;
+using namespace std::chrono_literals;
+
+namespace {
+
+// Verifies that using the MONITOR command works properly.
+// Opens a connection, issues a MONITOR, issues some commands to
+// generate some traffic, and waits for several MONITOR messages to arrive.
```

Comment:
> // Opens a connection, issues a MONITOR, issues some commands to
> // generate some traffic, and waits for several MONITOR messages to arrive.

I thought it was not allowed to issue commands on the same connection one MONITOR mode is entered. Perhaps this was only a restriction in RESP2 since it does not support server pushes.